modules/claims_api/app/controllers/concerns/claims_api/disability_compensation_validations.rb

```diff
@@ -172,11 +172,11 @@ def validate_form_526_location_codes!
       form_attributes['serviceInformation']['servicePeriods'].each do |service_period|
         next if Date.parse(service_period['activeDutyEndDate']) <= Time.zone.today
         next if separation_locations.any? do |location|
-                  location[:id] == service_period['separationLocationCode']
+                  location[:id]&.to_s == service_period['separationLocationCode']
```

Comment:
nit: we don't necessarily need `&` here since `nil.to_s` evaluates to the empty string.